### DISCUSSION TITLE
Any repo to get the program/code to install on docker?

### DISCUSSION THREAD

**Discussion by ChiFungLam** [iptv-org/community#945] [Q&A]
### Description

I would like to install the program/code onto some free docker like replit so that I can generate epg from the docker. But don't know the link to the repo of those code, is that github.com/iptv-org/epg.git ? (I am new to those free docker service). Thanks.

Regds
LAM Chi-fung

**Comment by octavioj** [comment]
One has to build the image to get this done. I am trying to do it and it works to fetch the data but I am still having a hard time getting it to start the webserver.

**Comment by octavioj** [comment]
> ### Description
> I would like to install the program/code onto some free docker like replit so that I can generate epg from the docker. But don't know the link to the repo of those code, is that github.com/iptv-org/epg.git ? (I am new to those free docker service). Thanks.
> 
> Regds LAM Chi-fung

I was able to put this into a container but there is something wrong with the generation of the XMLTV that only yields on channel. Here is the Dockerfile that uses one of the standard Node JS images, the one based on alpine so it uses less resources.

Dockerfile:

FROM node:current-alpine3.17

RUN apk update && \
    apk add --no-cache bash && \
    apk add git && \
    adduser -D -s /bin/bash frenchtvguide && \
    mkdir /opt/frenchtvguide && \
    chown frenchtvguide /opt/frenchtvguide

USER frenchtvguide

WORKDIR /opt/frenchtvguide/

RUN git clone --depth 1 -b master https://github.com/iptv-org/epg.git && \
    cd epg && \
    npm install

COPY start.sh /home/frenchtvguide/start.sh

EXPOSE 3000

CMD [ "bash" , "/home/frenchtvguide/start.sh" ]

Here is the start.sh file that creates the XMLTV and the corn job:

#!/bin/bash

cd /home/frenchtvguide/

npm --prefix /opt/frenchtvguide/epg run grab -- --site=chaines-tv.orange.fr --lang=fr && \
npm --prefix /opt/frenchtvguide/epg run grab -- --site=chaines-tv.orange.fr --cron="@hourly" --lang=fr & \
npm --prefix /opt/frenchtvguide/epg run serve

These are for the site chaines-tv.orange.fr . I will work on making it generic and use environment variables to select the corn expression as well as the site from which to pull the XMLTV. I hope this helps.

**Comment by weirdal3333** [comment]
I would like to have an docker image for this so i can integrate it with my jellyfin server somehow. Sometimes I like to distrohop, and having all of my services in docker images makes it really easy... I recently switched from Manjaro, for example.

This would probably help people with NAS systems too!

**Reply by octavioj** [reply]
I am still playing around with the docker build process and as of yesterday the build no longer works. I did not have time to fully troubleshoot the issue. I tried building under Apline and Debian images and got the same problem from the source.

**Comment by johnhelt** [comment]
I took the liberty of taking the above dockerfile, but changed the base image to one that contains both python and node and runs on alpine:
Here's a docker file that works: 

```
FROM nikolaik/python-nodejs:python3.11-nodejs20-alpine

RUN apk update && apk add --no-cache bash && apk add git && adduser -D -s /bin/bash tvguide && mkdir /opt/tvguide && chown tvguide /opt/tvguide

USER tvguide

WORKDIR /opt/tvguide/

RUN git clone --depth 1 -b master https://github.com/iptv-org/epg.git && cd epg && npm install

COPY start.sh /home/tvguide/start.sh

EXPOSE 3000

CMD [ "bash" , "/home/tvguide/start.sh" ]
```

the start.sh is:

```
#!/bin/bash
#Here is the start.sh file that creates the XMLTV and the corn job:

cd /home/tvguide/

npm --prefix /opt/tvguide/epg run grab -- --site=$SITE --lang=en &&
npm --prefix /opt/tvguide/epg run grab -- --site=$SITE --cron="@hourly" --lang=en &
npm --prefix /opt/tvguide/epg run serve
```

and if you want to use docker-compose here's the yaml:

```
version: '3.4'

services:
  iptvepg:
    build: .
    container_name: iptvepg
    ports:
      - 7090:3000
#      - 7096:7096 # Only for SSL
    restart: unless-stopped
    environment:
     - SITE

```
Note, I'm running other services on port 3000, so I'm mapping another port to that. You can set the SITE variable in an .env file, or using the docker-compose yaml directly.

**Comment by tohenk** [comment]
Here is the docker compose I've used, https://github.com/tohenk/docker-epg.

**Comment by freearhey** [comment]
Instructions on how to use it with Docker have been added to the description of the repository: https://github.com/iptv-org/epg?tab=readme-ov-file#docker
